woocommerce/wc-admin
# Use Rule Processing for Free Extensions List

Fixes https://github.com/woocommerce/woocommerce-admin/issues/6885

The Free Extensions list tab of the Business Details Step needs to be powered by remote JSON feed. This PR creates the endpoint in the same fashion that was done for Remote Payment Recommendations in https://github.com/woocommerce/woocommerce-admin/pull/6830.

This PR utilizes the [Remote Payments Tester](https://github.com/joshuatf/remote-payments-tester/pull/1) to provide a JSON feed because it doesn't yet exist on woocommerce.com.

### Screenshots

<img width="852" alt="Screen Shot 2021-05-05 at 2 05 50 PM" src="https://user-images.githubusercontent.com/1922453/117089940-09c74800-adab-11eb-90cb-5659ac02ce04.png">

### Detailed test instructions:

1. Install and activate the [Remote Payments Tester](https://github.com/joshuatf/remote-payments-tester/pull/1) using the branch in the PR if it hasn't been merged yet.
2. Take a look at the [ruleset in the data feed](https://github.com/joshuatf/remote-payments-tester/blob/2078bcdf004b392a6c33610a156215a48441a865/free-extensions.json) and see the countries allowed for WC Pay.
3. In Postman create a GET request to `/wp-json/wc-admin/onboarding/free-extensions` and see the resulting array of free plugins.
4. Confirm `is_visible` correctly reflects if your dev environment's base location country is in the allowed array of countries found on the feed.

### Next Steps

The full configuration as its implemented in the client can be seen here:

https://github.com/woocommerce/woocommerce-admin/blob/47bfc357258a9f74d361a6348597eb046699a0d9/client/profile-wizard/steps/business-details/flows/selective-bundle/selective-extensions-bundle/index.js#L56

1. Extend the `option` rule to look into nested object structures because industries are nested in the `woocommerce_onboarding_profile` option object.
2. Add other plugins to the config.
